src/lib.rs

```diff
@@ -1,3 +1,86 @@
+//! Welcome to the Momento Rust SDK!
+//!
+//! This crate contains a `CacheClient` for interacting with a serverless Momento Cache, and a
+//! `TopicClient` for interacting with serverless Momento Topics (pub/sub messaging). For more detailed
+//! information see the [Momento documentation](https://docs.momentohq.com).
+//! 
+//! Construct instances of these clients using `CacheClient::builder()` and `TopicClient::builder()`.
+//!
+//! A few conventions you will find in the SDK that are worth knowing about:
+//!
+//! ## Asynchronous APIs
+//!
+//! All APIs in the SDK are asynchronous and return `Future`s. This means that you will need to use an
+//! async runtime; we recommend [`tokio`](https://tokio.rs). Examples that include this dependency
+//! and illustrate the use of `#[tokio::main]` can be found in
+//! [the `example` directory of the github repo](https://github.com/momentohq/client-sdk-rust/tree/main/example).
+//!
+//!
+//! ## Configuration
+//! 
+//! Pre-built configurations are provided, with settings such as timeouts and keep-alives tuned
+//! to appropriate values for different environments. For example:
+//! 
+//! - `momento::cache::configurations::Laptop::latest()` - suitable for a development environment with lenient timeouts
+//! - `momento::cache::configurations::InRegion::latest()` - suitable for a production configuration with more strict timeouts.
+//! 
+//! These configurations can be passed to the `CacheClient` and `TopicClient` builders.
+//! For advanced use cases you can build your own configurations rather than using the pre-builts.
+//!
+//! ## Credential Providers
+//! 
+//! The `CredentialProvider` struct is used to provide the API key for the Momento service. The two
+//! most common factory functions for creating a `CredentialProvider` are:
+//!
+//! - `CredentialProvider::from_env_var` - reads the API key from an environment variable
+//! - `CredentialProvider::from_string` - takes the API key as a string; can be used when retrieving the key from a secret manager, etc.
+//!
+//! ## Error Handling
+//! 
+//! Most APIs return a `MomentoResult`, which is just a type alias for `Result<T, MomentoError>`. You
+//! can use a `match` statement to handle the `Result` or use the `?` operator to propagate errors.
+//!
+//! ## Enum Response Types, Type Coercion via `into` and `try_into`
+//! 
+//! Many APIs may have more than one type of response that they can return. For example, `CacheClient::get`
+//! may return a cache hit or a cache miss. These response are represented as enums, which you can
+//! interact with via a `match` statement, or you can use `try_into` to try to directly coerce the response
+//! into your desired type.
+//! 
+//! All Momento cache values are stored as `vec<u8>`, but if you are using UTF-8 strings, you can use `try_into`
+//! for these coercions as well.
+//! 
+//! Here are a few examples of how you can interact with a `CacheClient::get` response:
+//! 
+//! Using a `match`:
+//! 
+//! ```
```

Comment:
I think you need to specify `no_run` on these code blocks

Reply:
looks like you need to set up the whole doctest client if you want it to compile, or just `ignore` to make the doctests skip over it ([reference](https://stackoverflow.com/questions/32429369/how-to-opt-out-of-running-a-doc-test))

Reply:
yeah working on adding the client stuff, ty